### PR TITLE
management sdk readme

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

**Comment by realwanpengli** [comment]
Please review the readme, but ignore the links, since some links of samples are waiting for approvals.

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Azure SignalR Service Management SDK 
+
+## Build Status
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: Merge build status into README in root folder?

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  * Broadcast messages to all SignalR clients
+  * Publish messages to a specific user(s)
+  * Publish messages to a specific group(s)
+* Group membership:
+  * Add a specific user to a specific group
+  * Remove a specific user from a specific group
+
+#### Persistent Mode:
+
+Persistent mode includes the aboves fetures and are extended to all features that .Net Core SignalR supports.
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: What does "extended to all features that .Net Core SignalR supports" mean?

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    await hubContext.DisposeAsync();
+}
+```
+
+For full sample on how to use Management SDK can be found [here](TODO).
+
+## Transport Type
+
+This SDK can communicates to Azure SignalR Service with two transport types:
+* Transient: Create a Http request Azure SignalR Service for each message sent. It is useful when you are unable to establish a WebSockets connection.
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: Add a link to REST API doc

File: src/Microsoft.Azure.SignalR.Management/README.md (hunk 2 above)
Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR.Management/README.md]: Including all features in Transient mode, but supports more, such as: 
* broadcast except some connections
* send to connections
...

 "extended to all features that .Net Core SignalR supports"  sounds user unfriendly.
I will list all supported features instead.

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+Azure SignalR Service Management SDK helps you to manages SignalR clients through Azure SignalR Service directly. Therefore, this SDK can be but not limited to be used in [serverless](https://azure.microsoft.com/zh-cn/solutions/serverless/) environments. You can use this SDK to manage SignalR clients connected to your Azure SignalR Service in any environment, such as in a console app, in an Azure function or in an App Server.
+
+### Features
+
+#### Transient Mode:
+
+Features are limited. The supported features are:
+
+* Publishing messages:
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: How about make a table to compare the features between two modes?

File: README.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ Package Name | Target Framework | NuGet | MyGet
...
 Microsoft.Azure.SignalR.Protocols | .NET Standard 2.0 | [![NuGet](https://img.shields.io/nuget/v/Microsoft.Azure.SignalR.Protocols.svg)](https://www.nuget.org/packages/Microsoft.Azure.SignalR.Protocols) | [![MyGet](https://img.shields.io/myget/azure-signalr-dev/vpre/Microsoft.Azure.SignalR.Protocols.svg)](https://www.myget.org/feed/azure-signalr-dev/package/nuget/Microsoft.Azure.SignalR.Protocols)
 Microsoft.Azure.SignalR.AspNet | .NET Standard 2.0 | [![NuGet](https://img.shields.io/nuget/v/Microsoft.Azure.SignalR.AspNet.svg)](https://www.nuget.org/packages/Microsoft.Azure.SignalR.AspNet) | [![MyGet](https://img.shields.io/myget/azure-signalr-dev/vpre/Microsoft.Azure.SignalR.AspNet.svg)](https://www.myget.org/feed/azure-signalr-dev/package/nuget/Microsoft.Azure.SignalR.AspNet)
+Microsoft.Azure.SignalR.Management | .NET Standard 2.0 | null | [![MyGet](https://img.shields.io/myget/azure-signalr-dev/vpre/Microsoft.Azure.SignalR.Management.svg)](https://www.myget.org/feed/azure-signalr-dev/package/nuget/Microsoft.Azure.SignalR.Management)
```
Comment (chenkennt) [inline README.md]: Also add a brief intro to management sdk in this file and link to the new readme

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+``` C#
+var clientEndpoint = serviceManager.GetClientEndpoint("<Your Hub Name>");
+var accessToken = serviceManager.GenerateClientAccessToken("<Your Hub Name>", "<Your User ID>");
+```
+
+
+### Create and Use ServiceHubContext
+
+You can create a instance of `IServiceHubContext` to publish messages or manage group membership. The sample on how to use Management SDK to publish messages to SignalR clients can be found [here](https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management).
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: a -> an

File: src/Microsoft.Azure.SignalR.Management/README.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+                    })
+                    .Build();
+    
+
+```
+
+You can generate the endpoint for SignalR clients connecting to your Azure SignalR Service.
+You can also generate the access token for SignalR clients connecting to your Azure SignalR Service in serverless mode.
+
+Both of endpoint and access token are useful when you want to redirect SignalR clients to your Azure SignalR Service. The sample on how to use Management SDK to redirect SignalR clients to Azure SignalR Service can be found [here](https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management).
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Management/README.md]: Explain a bit more about why you need to implement a negotiate by yourself (as there is no web server in serverless scenario).

Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR.Management/README.md]: I think I have explained a little of it the next paragraph. The key point is the first sentence that connection string should be protected.

> Usually you don't want to expose your connection string to SignalR clients. In this situation, the SignalR clients connect to a specific endpoint which will return a special negotiation response and redirect clients to your Azure SignalR Service from the connection string. Read more details about the redirection at SignalR's [Negotiation Protocol](https://github.com/aspnet/SignalR/blob/master/specs/TransportProtocols.md#post-endpoint-basenegotiate-request).

That may not clear enough.

How about this, just move one sentence:
- - - - -
Usually you don't want to expose your connection string to SignalR clients. In this situation, the SignalR clients connect to a specific endpoint which will return a special negotiation response and redirect clients to your Azure SignalR Service from the connection string. 

Both of endpoint and access token are useful when you want to redirect SignalR clients to your Azure SignalR Service. The sample on how to use Management SDK to redirect SignalR clients to Azure SignalR Service can be found [here](https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management).


>  Read more details about the redirection at SignalR's [Negotiation Protocol](https://github.com/aspnet/SignalR/blob/master/specs/TransportProtocols.md#post-endpoint-basenegotiate-request).
- - - - -

Comment (chenkennt) [reply src/Microsoft.Azure.SignalR.Management/README.md]: I think it worth to call out in serverless scenario there is no server to do the negotiate so you need to host it by yourself. Also once you host the negotiate endpoint you can use SignalR client to connect as a normal hub.

**Pull request merged**